conda_devenv/devenv.py

```diff
@@ -0,0 +1,248 @@
+        if output_filename.endswith(".yml.yml"):
+            raise ValueError("Can't guess the output file, please provide the output file with the --output-filename "
+                             "flag")
+
+    if os.path.exists(output_filename) and not args.force:
+        raise ValueError("The output file '{}' already exists and the flag --force was omitted. Refusing to override "
+                         "the file.".format(output_filename))
+
+    with open(output_filename, 'w') as f:
+        f.write(rendered_contents)
+
+    return output_filename
+
+
+def __call_conda_env_update(args, output_filename):
+    import subprocess
+    command = [
+        "conda",
+        "env",
+        "update",
+        "--file",
+        output_filename,
+    ]
+    if not args.no_prune:
+        command.append("--prune")
+    if args.name:
+        command.extend(["--name", args.name])
+    print("> Executing: %s" % ' '.join(command))
+    return subprocess.call(command)
+
+
+def __write_activate_deactivate_scripts(args, conda_yaml_dict, environment):
+    import subprocess
+    env_name = args.name or conda_yaml_dict["name"]
+    conda_root = subprocess.check_output(["conda", "info", "--root"])
+    if sys.version_info >= (3, 0, 0):
+        conda_root = conda_root.decode("utf-8")
+    conda_root = conda_root.strip()
+
+    from os.path import join
+
+    env_directory = join(conda_root, "envs", env_name)
+
+    activate_script = render_activate_script(environment)
+    deactivate_script = render_deactivate_script(environment)
+
+    activate_directory = join(env_directory, "etc", "conda", "activate.d")
+    deactivate_directory = join(env_directory, "etc", "conda", "deactivate.d")
+
+    os.makedirs(activate_directory, exist_ok=True)
+    os.makedirs(deactivate_directory, exist_ok=True)
+
+    extension = ".bat" if sys.platform.startswith("win") else ".sh"
+    with open(join(activate_directory, "devenv-vars" + extension), "w") as f:
+        f.write(activate_script)
+    with open(join(deactivate_directory, "devenv-vars" + extension), "w") as f:
+        f.write(deactivate_script)
+
+
+def main():
+    parser = argparse.ArgumentParser(description="Work with multiple conda-environment-like yaml files in dev mode.")
+    parser.add_argument("--file", "-f", nargs="?", help="The environment.devenv.yml file to process.")
+    parser.add_argument("--name", "-n", nargs="?", help="Name of environment.")
+    parser.add_argument("--print", help="Only prints the rendered file to stdout and exits.", action="store_true")
+    parser.add_argument("--no-prune", help="Don't pass --prune flag to conda-env.", action="store_true")
+    parser.add_argument("--output-file", nargs="?", help="Output filename.")
+    parser.add_argument("--force", action="store_true", help="Overrides the output file, even if it already exists.")
+
+    args = parser.parse_args()
+
+    filename = args.file or "environment.devenv.yml"
```

Comment:
Put this default on the `--file` argument so it will appear on the command-line help automatically (I hope)